nifi-registry-core/nifi-registry-data-model/src/main/java/org/apache/nifi/registry/extension/ExtensionBundleVersionMetadata.java

```diff
@@ -0,0 +1,161 @@
+
+    @NotBlank
+    private String sha256Hex;
+
+
+    @ApiModelProperty(value = "The id of this version of the extension bundle")
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    @ApiModelProperty(value = "The id of the extension bundle this version is for")
+    public String getExtensionBundleId() {
+        return extensionBundleId;
+    }
+
+    public void setExtensionBundleId(String extensionBundleId) {
+        this.extensionBundleId = extensionBundleId;
+    }
+
+    @ApiModelProperty(value = "The id of the bucket the extension bundle belongs to", required = true)
+    public String getBucketId() {
+        return bucketId;
+    }
+
+    public void setBucketId(String bucketId) {
+        this.bucketId = bucketId;
+    }
+
+    @ApiModelProperty(value = "The version of the extension bundle")
+    public String getVersion() {
+        return version;
+    }
+
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    @ApiModelProperty(value = "The optional bundle dependency (i.e. another bundle this bundle is dependent on)")
+    public ExtensionBundleVersionDependency getDependency() {
+        return dependency;
+    }
+
+    public void setDependency(ExtensionBundleVersionDependency dependency) {
+        this.dependency = dependency;
+    }
+
+    @ApiModelProperty(value = "The timestamp of the create date of this version")
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    public void setTimestamp(long timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    @ApiModelProperty(value = "The identity that created this version")
+    public String getAuthor() {
+        return author;
+    }
+
+    public void setAuthor(String author) {
+        this.author = author;
+    }
+
+    @ApiModelProperty(value = "The description for this version")
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    @ApiModelProperty(value = "The hex representation of the SHA-256 digest of the binary content for this version")
+    public String getSha256Hex() {
+        return sha256Hex;
+    }
+
+    public void setSha256Hex(String sha256Hex) {
+        this.sha256Hex = sha256Hex;
+    }
+
+    @Override
+    public int compareTo(final ExtensionBundleVersionMetadata o) {
+        return o == null ? -1 : version.compareTo(o.version);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.id);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+
+        final ExtensionBundleVersionMetadata other = (ExtensionBundleVersionMetadata) obj;
+        return Objects.equals(this.id, other.id);
```

Comment:
Will this cause issues with `Set`/`List` assignments/`#contains()` because the IDs can be equal but the data within them different? I remember this was always an issue with Java O/RM.

Reply:
Also, unfortunately we have no ORM here and are writing our own JDBC :(

Reply:
Works as expected.